Sources/AblyChat/Messages.swift

```diff
@@ -0,0 +1,74 @@
+import Ably
+
+public protocol Messages: AnyObject, Sendable, EmitsDiscontinuities {
+    func subscribe(bufferingPolicy: BufferingPolicy) -> MessageSubscription
+    func get(options: QueryOptions) async throws -> any PaginatedResult<Message>
+    func send(params: SendMessageParams) async throws -> Message
+    var channel: ARTRealtimeChannelProtocol { get }
+}
+
+public struct SendMessageParams: Sendable {
+    public var text: String
+    public var metadata: MessageMetadata?
+    public var headers: MessageHeaders?
+
+    public init(text: String, metadata: MessageMetadata? = nil, headers: MessageHeaders? = nil) {
+        self.text = text
+        self.metadata = metadata
+        self.headers = headers
+    }
+}
+
+public struct QueryOptions: Sendable {
+    public enum Direction: Sendable {
+        case forwards
+        case backwards
+    }
+
+    public var start: Date?
+    public var end: Date?
+    public var limit: Int?
+    public var direction: Direction?
+
+    public init(start: Date? = nil, end: Date? = nil, limit: Int? = nil, direction: QueryOptions.Direction? = nil) {
+        self.start = start
+        self.end = end
+        self.limit = limit
+        self.direction = direction
+    }
+}
+
+public struct QueryOptionsWithoutDirection: Sendable {
```

Comment:
what would the default direction be?

Reply:
`getPreviousMessages` returns messages newest-to-oldest; opened https://github.com/ably/ably-chat-js/pull/328 to document this in JS

Reply:
@umair-ably anything to do here? if not could you resolve please?

Reply:
all good :)